dali/test/python/test_external_source_parallel_custom_serialization.py

```diff
@@ -305,7 +301,7 @@ def _test_pickle_by_value_decorator_on_global_function(name, py_callback_pickler
 
 
 @register_case(tests_dali_pickling)
-@raises(NoDumpsParam)
+@raises_pattern(ValueError, "Expected special_dumps_param among kwargs, got *")
```

Comment:
General comment, do we want to make the name shorter? 
Would we be fine with just `raises`?
Can we make the API something like:
```
raises(exception_type, glob=None, regex=None, case_sensitive=False)
```
That way, if you want the glob, you do
```
@raises(ValueError, "My * error")
```
for regex, you do:
```
@raises(ValueError, regex="My .*ror")
```

How easy it is to pass a compiled regex? Shouldn't we just allow the regex string, as it is usually defined above the tests in the global scope, so we don't want to deal with adding re.compile("pattern") boilerplate to the match.

Reply:
It allows using regex as string, compiled patterns are just passed through (because why not, maybe there will be some use case for special hand-crafted pattern compiled with extra flags).

For now this is how using regex looks like (this builds with case-insensitive flag):

```
@raises_pattern(ValueError, "My [a-z]*\\s[0-9]{2} message", use_glob=False)
def test_regex_insensitive():
    raise ValueError("my ahgsfds 42 Message")
```

And manually built pattern:

```
import re
pattern = re.compile("[a-z]{3}[0-9]+")

@raises_pattern(ValueError, pattern)
def test_precompiled_regex():
    raise ValueError("abc42")
```

Reply:
Ah, ok, I misread. Still, I think, `glob` and `regex` patterns would be probably easier than the boolean to switch between them?

Reply:
@stiepan  Why do we need regex strings? We could just do:
```
@raises(ValueError, r"lorem ipsum")
```
which would match all strings that contain `lorem ipsum`.

Reply:
@mzient yep, you can use r"" literals too.

Reply:
We should consider what's the most common use case and make that shortest and easiest to use. For example, if the most common use case is to check for a presence of a string inside the message, then we should have sth like:
```@raises(ErrorType, "part of the message")```
and to supplement that:
```Python
@raises(ErrorType, regex="^Could not allocate.*bytes")```
```Python
@raises(ErrorType, glob=Could not allocate*bytes*")```

Reply:
@mzient 

```
@raises(ErrorType, "part of the message")
```
is now by default
@raises(ErrorType, glob="part of the message") that will search for any occurrence inside the message.

To enforce matching the whole message, one could use regex with beginning/end of string explicitly stipulated.